### PR TITLE
Added exponential timeout depending on the current round with a cap.

### PR DESCRIPTION
**Pull request by 0xSasaPrsic**
Implementation of timeout function with a cap in order not to get stuck in RoundChangeState when the timeout is too large.

### CODE REVIEW

File: consensus_test.go [hunk 1]
```diff
@@ -465,8 +465,8 @@ func (m *mockPbft) emitMsg(msg *MessageReq) {
 
 func (m *mockPbft) addMessage(msg *MessageReq) {
 	// convert the address from the address pool
-	//from := m.pool.get(string(msg.From)).Address()
-	//msg.From = from
+	// from := m.pool.get(string(msg.From)).Address()
```
Comment (ferranbt) [inline consensus_test.go:468]: Why is this being commented?

File: consensus.go [hunk 2]
```diff
@@ -683,16 +683,14 @@ func (p *Pbft) forceTimeout() {
...
-		timeout = inputTimeout[0]
-	}
-
+// exponentialTimeout calculates the timeout duration depending on the current round
+func (p *Pbft) exponentialTimeout() time.Duration {
+	timeout := defaultTimeout
 	round := p.state.view.Round
-	if round > 0 {
-		timeout += time.Duration(math.Pow(2, float64(round))) * time.Second
+	if round <= 8 {
```
Comment (ferranbt) [inline consensus.go:690]: Can you use `maxTimeout` here?

**Review by ferranbt — CHANGES_REQUESTED** [review]

File: consensus_test.go (hunk 1 above)
Comment (sprsic) [reply consensus_test.go:468]: This was already commented on in the main branch, and I did only formatting comments. Should I revert formatting?

File: consensus.go (hunk 2 above)
Comment (0xSasaPrsic) [reply consensus.go:693]: Added a comment here about for <=8 as discussed.

**Pull request merged**